Source/WebKit/UIProcess/AuxiliaryProcessProxy.cpp

```diff
@@ -186,7 +186,7 @@ void AuxiliaryProcessProxy::terminate()
 {
     RELEASE_LOG(Process, "AuxiliaryProcessProxy::terminate: PID=%d", processID());
 
-#if PLATFORM(COCOA)
+#if PLATFORM(COCOA) && !USE(EXTENSIONKIT)
```

Comment:
I am a little confused why this function doesn't do anything when USE(EXTENSIONKIT). I don't think `processLauncher->terminateProcess();` is enough as the process may not be launching anymore.

Reply:
That is a good point! In the ExtensionKit case, I believe `processLauncher->terminateProcess()` is sufficient, since it will call invalidate on the process object, and thereby terminate the process.

Reply:
I have to think about this. It seems confusing to 100% rely on the process launcher to terminate a process. While it seems OK to involve the process launcher WHILE a process is launching, it is more surprising to do so after it's finished launching.

Reply:
The previous code was trying to kill the process and would only fall back to using the process launcher. The new code doesn't feel as nice, which is unfortunate since it is supposed the new modern code.

Reply:
That is a good point, Chris! I wonder if we should make it more explicit that we hold a reference to the Process launcher throughout the lifetime of the process. Perhaps we could make it a data member instead of having it as a reference data member in AuxiliaryProcessProxy? We can look into this in a follow-up :)

Thanks for reviewing!

Reply:
Right, let's not block fixing the bug. We should land this and we can revisit later if we have something nicer.

Reply:
Yes, that sounds good. Thanks for approving!